pkg/agent/cniserver/server.go

```diff
@@ -521,6 +552,49 @@ func (s *CNIServer) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// interceptAdd handles Add request in policy only mode. Another CNI must already
+// be called prior to Antrea CNI to allocate IP and ports. Antrea takes allocated port
+// and hooks it to OVS br-int.
+func (s *CNIServer) interceptAdd(cniConfig *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	klog.Infof("CNI Chaining: add")
+	prevResult, response := s.parsePrevResultFromRequest(cniConfig.NetworkConfig)
+	if response != nil {
+		klog.Infof("Failed to parse prev result for container %s.", cniConfig.ContainerId)
+		return response, nil
+	}
+	podName := string(cniConfig.K8S_POD_NAME)
+	podNamespace := string(cniConfig.K8S_POD_NAMESPACE)
+	result := make([]byte, 0, 0)
+	if err := s.podConfigurator.connectInterceptedInterface(
+		podName,
+		podNamespace,
+		cniConfig.ContainerId,
+		s.hostNetNsPath(cniConfig.Netns),
+		cniConfig.Ifname,
+		prevResult.IPs); err != nil {
+		return &cnipb.CniCmdResponse{CniResult: result}, fmt.Errorf("failed to connect container %s to ovs: %w", cniConfig.ContainerId, err)
+	}
+	// Notify the Pod update event to required components.
+	s.podUpdates <- v1beta1.PodReference{Name: podName, Namespace: podNamespace}
+
+	return &cnipb.CniCmdResponse{CniResult: cniConfig.NetworkConfiguration}, nil
+}
+
+func (s *CNIServer) interceptDel(cniConfig *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	// wait for patch port to be configured
+	klog.Infof("CNI Chaining: delete")
+	return &cnipb.CniCmdResponse{CniResult: make([]byte, 0, 0)}, s.podConfigurator.disconnectInterceptedInterface(
+		string(cniConfig.K8S_POD_NAME),
+		string(cniConfig.K8S_POD_NAMESPACE),
+		cniConfig.ContainerId)
+}
+
+func (s *CNIServer) interceptCheck(_ *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	klog.Infof("CNI Chaining: check")
+	// TODO, check for host interface setup later
```

Comment:
will this be in a follow up PR? or this PR itself

Reply:
perhaps in foreseeable future, this only make sense if Antrea wants to do some additional checking ...